### PR TITLE
pageserver: fix treating all download errors as 'Other'

### PR DESCRIPTION
**Pull request by jcsp**
## Problem

`download_retry` correctly uses a fatal check to avoid retrying forever on cancellations and NotFound cases.  However, `download_layer_file` was casting all download errors to "Other" in order to attach an anyhow::Context.

Noticed this issue in the context of secondary downloads, where requests to download layers that might not exist are issued intentionally, and this resulted in lots of error spam from retries that shouldn't have happened.

## Summary of changes

- Remove the `.context()` so that the original DownloadError is visible to backoff::retry

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: pageserver/src/tenant/remote_timeline_client/download.rs [hunk 1]
```diff
@@ -98,9 +90,11 @@ pub async fn download_layer_file<'a>(
 
             let bytes_amount = tokio::io::copy_buf(&mut reader, &mut destination_file)
                 .await
-                .with_context(|| format!(
+                .with_context(|| {
```
Comment (jcsp) [inline pageserver/src/tenant/remote_timeline_client/download.rs:93]: This is an unrelated whitespace change that rustfmt has made as a side effect of removing the with_context() wrapper above.

**Review by koivunej — APPROVED** [review]
Thanks, I failed to consider this.

File: pageserver/src/tenant/remote_timeline_client/download.rs (hunk 1 above)
Comment (jcsp) [reply pageserver/src/tenant/remote_timeline_client/download.rs:93]: Yep, agree that there's still an issue for cancellation during the body receive -- would be good to tackle that in a separate PR.
